packages/rum/src/domain/record/observers.ts

```diff
@@ -371,9 +371,41 @@ function initStyleSheetObserver(cb: StyleSheetRuleCallback): ListenerHandler {
     },
   })
 
+  const { stop: restoreInsertNestedRule } = instrumentMethodAndCallOriginal(CSSGroupingRule.prototype, 'insertRule', {
+    before(rule, index) {
+      if (hasSerializedNode(this.parentStyleSheet!.ownerNode!)) {
+        cb({
+          id: getSerializedNodeId(this.parentStyleSheet!.ownerNode),
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏factorize a bit with something like

```ts
function withStyleSheetId(styleSheet: StyleSheet | null, callback: (id: number) => {
  if (styleSheet && hasSerializeNode(styleSheet.ownerNode)) {
    callback(getSerializedNode(styleSheet.ownerNode)
  }
})

// usage
withStyleSheetId(this.parentStyleSheet, id => cb({ id, adds: [...] )})
```